package/MDAnalysis/coordinates/memory.py

```diff
@@ -352,8 +352,11 @@ def timeseries(self, asel=None, start=0, stop=-1, step=1, format='afc'):
         # Return a view if either:
         #   1) asel is None
         #   2) asel corresponds to the selection of all atoms.
+        # The 2nd case may need a deep comparison because the "all" selection
+        # creates a new AtomGroup object that isn't the same as universe.atoms.
         array = array[basic_slice]
-        if (asel is None or asel is asel.universe.atoms):
+        if (asel is None or asel is asel.universe.atoms or
+                np.array_equal(asel.indices, asel.universe.atoms.indices)):
```

Comment:
The `array_equal` check here will scale with system size as it will check (potentially) all array elements.  Is it necessary to include it?